src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationOptions.cs

```diff
@@ -110,16 +134,38 @@ public SuperOfficeAuthenticationEnvironment Environment
         /// </remarks>
         public TimeSpan JwksCacheLifetime { get; set; } = TimeSpan.FromMinutes(15);
 
+        /// <summary>
+        /// 1 day is the default time interval that afterwards, <see cref="ConfigurationManager" /> will obtain new configuration.
+        /// </summary>
+        public TimeSpan AutomaticRefreshInterval { get; set; } = TimeSpan.FromDays(1);
+
+        /// <summary>
+        /// The minimum time between <see cref="ConfigurationManager" /> retrievals, in the event that a retrieval failed, or that a refresh was explicitly requested. 30 seconds is the default.
+        /// </summary>
+        public TimeSpan RefreshInterval { get; set; } = TimeSpan.FromSeconds(30);
+
+        /// <summary>
+        /// Gets or sets the parameters used to validate identity tokens.
+        /// </summary>
+        /// <remarks>Contains the types and definitions required for validating a token.</remarks>
+        public TokenValidationParameters TokenValidationParameters { get; set; } = new TokenValidationParameters();
+
         /// <inheritdoc />
         public override void Validate()
         {
             base.Validate();
 
             if (_environment == SuperOfficeAuthenticationEnvironment.Production
-                && !AuthorizationEndpoint.StartsWith("https://", System.StringComparison.InvariantCultureIgnoreCase))
+                && !AuthorizationEndpoint.StartsWith("https://", System.StringComparison.OrdinalIgnoreCase))
             {
                 throw new NotSupportedException("Production environment requires secure endpoints, i.e. begins with 'https://'.");
             }
+
+            if (ConfigurationManager == null)
+            {
+                throw new InvalidOperationException($"Provide {nameof(Authority)}, {nameof(MetadataAddress)}, "
+                + $"or {nameof(ConfigurationManager)} to {nameof(SuperOfficeAuthenticationOptions)}");
```

Comment:
```suggestion
                throw new InvalidOperationException(
                    $"Provide {nameof(Authority)}, {nameof(MetadataAddress)}, or {nameof(ConfigurationManager)} to {nameof(SuperOfficeAuthenticationOptions)}.");
```

Reply:
Editor forced this when on multiple lines. I've updated it to a one-liner.